docs/conceptual/More-about-how-ROCm-uses-PCIe-Atomics.rst

```diff
@@ -1,156 +1,63 @@
 
-The full list of HSA system architecture platform requirements are here:
-`HSA Sys Arch Features <http://hsafoundation.com/wp-content/uploads/2021/02/HSA-SysArch-1.2.pdf>`_.
+ROCm uses platform atomics to:
 
-AMD ROCm Software uses the new PCI Express 3.0 (Peripheral Component Interconnect Express [PCIe]
-3.0) features for atomic read-modify-write transactions which extends inter-processor synchronization
-mechanisms to IO to support the defined set of HSA capabilities needed for queuing and signaling
-memory operations.
+* Update the HSA queue's ``read_dispatch_id``. The command processor on the GPU agent uses a 64-bit atomic add operation. It updates the packet ID it processed.
+* Update the HSA queue's ``write_dispatch_id``. The CPU and GPU agents use a 64-bit atomic add operation. It supports multi-writer queue insertions.
+* Update HSA Signals. A 64-bit atomic operation is used for CPU & GPU synchronization.
 
-The new PCIe atomic operations operate as completers for ``CAS`` (Compare and Swap), ``FetchADD``,
-``SWAP`` atomics. The atomic operations are initiated by the I/O device which support 32-bit, 64-bit and
-128-bit operand which target address have to be naturally aligned to operation sizes.
 
-For ROCm the Platform atomics are used in ROCm in the following ways:
+PCIe for atomic operations
+----------------------------
+ROCm requires CPUs that support PCIe atomics. Similarly, all connected I/O devices should also support PCIe atomics for optimum compatibility. PCIe supports the ``CAS`` (Compare and Swap), ``FetchADD``, and ``SWAP`` atomic operations across multiple resources. These atomic operations are initiated by the I/O devices that support 32-bit, 64-bit, and 128-bit operands. Likewise, the target memory address where these atomic operations are performed should also be aligned to the size of the operand. This alignment ensures that the operations are performed efficiently and correctly without failure. 
 
-  * Update HSA queue's read_dispatch_id: 64 bit atomic add used by the command processor on the
-    GPU agent to update the packet ID it processed.
-  * Update HSA queue's write_dispatch_id: 64 bit atomic add used by the CPU and GPU agent to
-    support multi-writer queue insertions.
-  * Update HSA Signals -- 64bit atomic ops are used for CPU & GPU synchronization.
+When an atomic operation is successful, the requester receives a response of completion along with the operation result. However, any errors associated with the operation are signaled to the requester by updating the Completion Status field. Issues accessing the target location or running the atomic operation are common errors. Depending upon the error, the Completion Status field is updated to Completer Abort (CA) or Unsupported Request (UR). The field is present in the Completion Descriptor.
```

Comment:
Who is the intended audience? This information seems irrelevant to people writing ROCm applications. This is going into details of the PCIe bus protocol for dealing with error conditions.

Reply:
This falls under the conceptual section, and the audience can be someone using the document to troubleshoot the error conditions as well.